src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,124 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int DEFAULT_SIZE = 0;
+    private int quantity = DEFAULT_SIZE;
+    private Object[] elements = new Object[DEFAULT_CAPACITY];
+
     @Override
     public void add(T value) {
-
+        growIfArrayFull();
+        elements[quantity] = value;
+        quantity += 1;
     }
 
     @Override
     public void add(T value, int index) {
-
+        checkInRangeForAddition(index);
+        growIfArrayFull();
+        Object [] temp = new Object[elements.length];
+        System.arraycopy(elements, 0, temp, 0, temp.length);
```

Comment:
This is a redundant variable and unnecessary array copying. You are creating a new array and copying all elements from the old array to it, but you don't actually need to do this. You can just shift the elements to the right of the index where you want to insert the new element. This will make your code more efficient.